lib/web_ui/lib/src/engine/js_interop/js_promise.dart

```diff
@@ -11,40 +11,27 @@ import 'package:js/js_util.dart' as js_util;
 
 import '../util.dart';
 
-@JS()
-@staticInterop
-class PromiseResolver<T extends Object?> {}
-
-extension PromiseResolverExtension<T extends Object?> on PromiseResolver<T> {
-  void resolve(T result) => js_util.callMethod(this, 'call', <Object>[this, if (result != null) result]);
-}
-
-@JS()
-@staticInterop
-class PromiseRejecter {}
-
-extension PromiseRejecterExtension on PromiseRejecter {
-  void reject(Object? error) => js_util.callMethod(this, 'call', <Object>[this, if (error != null) error]);
+extension CallExtension on JSFunction {
+  external void call(JSAny? this_, JSAny? object);
 }
 
-/// Type-safe JS Promises
 @JS('Promise')
-@staticInterop
-abstract class Promise<T extends Object?> {
-  /// A constructor for a JS promise
-  external factory Promise(PromiseExecutor<T> executor);
-}
+external JSAny get _promiseConstructor;
+
+JSPromise createPromise(JSFunction executor) =>
```

Comment:
Why not keep the old interface and have `Promise` take in a `JSFunction` instead?

Reply:
I really was annoyed by our redundant declarations of types that already exist in `dart:js_interop`. We actually redeclared the promise type twice. Blech. Ideally I would just declare an extension method, but you can't do that for factories, so I either have to use `js_util` to call the constructor directly (as I've done here), or I have to redeclare the class and then cast it to `JSPromise`. Both seem kind of gross.

Reply:
Makes sense. The right fix for this is for me to add a constructor to `JSPromise`. If we could use extension types, then we could subtype `JSPromise` and use that to avoid the cast, but still not quite as good as the constructor.